iOS/MobileStarter/react-app/src/ProjectPicker.tsx

```diff
@@ -0,0 +1,143 @@
+async function getProjects(progress?: ProgressCallback, source = ProjectsSource.All, searchString = "") {
+  const client = new ProjectsAccessClient();
+  const chunkSize = 100;
+  const allProjects: Project[] = [];
+  const accessToken = await IModelApp.getAccessToken();
+
+  try {
+    let queryArgs: ProjectsQueryArg = { pagination: { skip: 0, top: chunkSize } };
+    if (source === ProjectsSource.All && searchString.length > 0)
+      queryArgs.search = { searchString, propertyName: ProjectsSearchableProperty.Name, exactMatch: false };
+    else
+      queryArgs.source = source;
+    const chunk = await client.getByQuery(accessToken, queryArgs);
+    allProjects.push(...chunk.projects);
+    progress?.({ loaded: allProjects.length });
+    return { projects: allProjects, next: chunk.links?.next };
+  } catch (ex) {
+    console.log(`Exception fetching projects: ${ex}`);
+    if (allProjects.length > 0) {
+      return { projects: allProjects, next: undefined };
+    } else {
+      throw ex;
+    }
+  }
+}
+
+interface ProjectButtonProps extends Omit<ButtonProps, "title"> {
+  project: Project;
+}
+
+function ProjectButton(props: ProjectButtonProps) {
+  const { project, onClick } = props;
+  return <HubScreenButton
+    title={project.name ?? ""}
+    onClick={onClick}
+  />;
+}
+
+interface ProjectListProps extends HubScreenButtonListProps {
+  projects: Project[];
+  onClick?: (project: Project) => void;
+}
+
+function ProjectList(props: ProjectListProps) {
+  const { projects, loading, onClick, children } = props;
+  return <HubScreenButtonList loading={loading}>
+    {projects.map((project, index) => <ProjectButton key={index} project={project} onClick={() => onClick?.(project)} />)}
+    {children}
+  </HubScreenButtonList>;
+}
+
+export interface ProjectPickerProps {
+  signedIn: boolean;
+  onSelect?: (project: Project) => void;
+  onError?: (error: any) => void;
+}
+
+export function ProjectPicker(props: ProjectPickerProps) {
+  const { signedIn, onSelect, onError } = props;
+  const [projects, setProjects] = React.useState<Project[]>([]);
+  const [search, setSearch] = React.useState("");
+  const [projectSource, setProjectSource] = React.useState(ProjectsSource.Recents);
+  const [loading, setLoading] = React.useState(false);
+  const [nextFunc, setNextFunc] = React.useState<ProjectsQueryFunction>();
+  const [loadingMore, setLoadingMore] = React.useState(false);
+  const loadMoreLabel = React.useMemo(() => i18n("HubScreen", "LoadMore"), []);
+  const isMountedRef = useIsMountedRef();
+  const searchLabel = React.useMemo(() => i18n("HubScreen", "Search"), []);
+  const projectSources = React.useMemo(() => [ProjectsSource.All, ProjectsSource.Recents, ProjectsSource.Favorites], []);
+  const projectSourceLabels = React.useMemo(() => [i18n("HubScreen", "All"), i18n("HubScreen", "Recents"), i18n("HubScreen", "Favorites")], []);
+
+  React.useEffect(() => {
+    if (!isMountedRef.current)
+      return;
+
+    if (!signedIn) {
```

Comment:
It seems like we should show an error in this case.

Reply:
I removed the signedIn prop, it isn't needed.